libraries/botbuilder-dialogs/src/dialogHelper.ts

```diff
@@ -49,46 +52,104 @@ export async function runDialog(
     }
 
     const dialogSet = new DialogSet(accessor);
-    dialogSet.telemetryClient = dialog.telemetryClient;
     dialogSet.add(dialog);
```

Comment:
I have some vague recollection of a side-effect of adding a dialog to a dialog set triggering telemetry client updates. Does this reordering break that side-effect?

Reply:
Thanks for pointing this out, I did some digging and this is what should happen. (I didn't step through the debugger)

### PR behavior

- Add dialog to dialogSet
  - No telemetryClient exists on dialogSet, so dialog.telemetryClient [stays the same](https://github.com/microsoft/botbuilder-js/blob/08c0e54e8788b826b337f826d0d60bf83dc7d735/libraries/botbuilder-dialogs/src/dialogSet.ts#L157-L160)
- Update DialogSet to use same telemetry client as dialog

Result: DialogSet and Dialog use the same TelemetryClient

### Original behavior

- Set dialogSet.telemetryClient to dialog.telemetryClient 
- Add dialog to dialogSet
  - dialogSet.telemetryClient exists, so set dialog.telemetryClient to use dialogSet.telemetryClient (same [link](https://github.com/microsoft/botbuilder-js/blob/08c0e54e8788b826b337f826d0d60bf83dc7d735/libraries/botbuilder-dialogs/src/dialogSet.ts#L157-L160) as above)
  - Since dialogSet uses dialog's telemetry client, an unnecessary setting occurs
  
 Result: DialogSet and Dialog use the same TelemetryClient

___

However, it seems like there may be a bug in the .NET implementation. If IBotTelemetryClient is found in the TurnState, this will update the telemetryClient on the DialogSet, but not the Dialog. @johnataylor